src/Button/Button.component.tsx

```diff
@@ -370,16 +373,20 @@ const StyledButton = styled.button<StyledButtonProps>`
                   min-width: ${minWidth || `${dimensions[$size].width}rem`};
               `}
 
-    ${({ icon, iconOnly, $size }: StyledButtonProps) =>
-        icon &&
+    ${({ iconOnly, $size }: StyledButtonProps) =>
         !iconOnly &&
         css`
             // Space icon from text
-            & > .anchor-icon {
+            & > .anchor-button-prefix {
                 margin-right: ${$size === 'lg' || $size === 'md'
```

Comment:
Since you're comparing against a string value (and in multiple places), might be a good idea to assign the strings to constants to avoid magic numbers. i.e. const LG = 'lg'

Reply:
I'm not sure this is really necessary because it's enforced by types, which are defined above. If I add consts for all these then I'm duplicating the information already outlined in the ButtonSize type.

Reply:
It has less to do with that it is enforced by types, and more to do with making a comparison in code, particularly because `'lg'` and `'md'` are used in multiple places. It's a logical expression and IMO, those should always be against a `const` and not an inline string literal (or number).

Reply:
Most of this will be reorganized when we switch to using a theme solution, so I'm going to punt on this and keep it in mind when that refactor happens.